### PR TITLE
remove the update action from theme update_extension schema

### PR DESCRIPTION
**Pull request by bahar-p**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/shopify/issues/437598
tl;dr: Update action is descoped from the project. Further context can be found [here](https://docs.google.com/document/d/1kfeB1SFg9oygW_0mEHON51eB4du_2QNq5US46mEJd_A/edit#heading=h.243m6mt93xx0)  

### WHAT is this pull request doing?

Removing the Update action from the update_extension schema. The zod schema should be consistent with core and [theme-liquid-docs](https://github.com/Shopify/theme-liquid-docs). Given I've already updated the schema in our liquid-docs ([PR](https://github.com/Shopify/theme-liquid-docs/pull/228)), I regenerated the schema here by running `pnpm run schema:generate` which in turn generates the schema based on the [theme-liquid-docs](https://github.com/Shopify/theme-liquid-docs). 

### How to test your changes?
Updated and ran the related test suite, which is `packages/theme/src/cli/services/update/check.test.ts` 

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

**Comment by bahar-p** [comment]
> If the changes are user-facing, run "pnpm changeset add" to track your changes and include them in the next release CHANGELOG.

@karreiro or @mgmanzella Given the feature isn't released yet, do we still need to do this?

**Review by karreiro — APPROVED** [review]
Thank you, @bahar-p! Great stuff 🚀 

> Given the feature isn't released yet, do we still need to do this [run `pnpm changeset add`] ?

+1 I don't think we need to update the `.changeset` because this is not a user-facing change indeed :)
